src/arch_switch.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * This software is licensed under the terms of the MIT-License
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2019, p-sam <p-sam@d3vs.net>.
+ */
+
+#include "arch_switch.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <switch/runtime/devices/socket.h>
+#include <switch/runtime/nxlink.h>
+#include <switch/services/applet.h>
+#include <switch/services/fs.h>
+
+#include "taisei.h"
+
+#define NX_LOG_FMT(fmt, ...) tsfprintf(stdout, "[NX] " fmt "\n", ##__VA_ARGS__)
+#define NX_LOG(str) NX_LOG_FMT("%s", str)
+#define NX_SETENV(name, val) NX_LOG_FMT("Setting env var %s to %s", name, val);env_set_string(name, val, true)
+
+static nxAtExitFn g_nxAtExitFn = NULL;
+static char g_programDir[FS_MAX_PATH] = {0};
+static AppletHookCookie g_hookCookie;
+
+static void onAppletHook(AppletHookType hook, void *param) {
+	switch (hook) {
+		case AppletHookType_OnExitRequest:
+			NX_LOG("Got AppletHook OnExitRequest, exiting.\n");
+			taisei_quit();
+			break;
+
+		default:
+			break;
+	}
+}
+
+attr_used
+void userAppInit() {
```

Comment:
Functions that take no parameters should be `(void)`; `()` means unspecified parameters.